teps/0098-workflows.md

```diff
@@ -256,11 +306,164 @@ Why should this TEP _not_ be implemented?
+built and released as a Tekton project, and likely renamed to "Workflows", since the scope
+of the Workflows project is larger than storing Pipelines in version control.
+The Tekton community would maintain the project and shape its future to meet the goals of the Workflows project.
+The doc [Pipelines as Code -> Tekton Workflows](https://docs.google.com/document/d/1JAnXFkvs4aeciAJEsl7454e0J3ZHbL1yCPncMBHlTtM/)
+compares Pipelines as Code to the goals of the Workflows project.
+
+Suggested steps:
+1. Design and implement a solution for parameter substitution in Pipelines as Code that's Tekton conformant.
+(The current [syntax](https://pipelinesascode.com/docs/guide/authoringprs/#authoring-pipelineruns-in-tekton-directory)
+for variable expansion isn't Tekton conformant.)
+1. Move the project to the tektoncd org, rename it, run tests on Tekton infrastructure, and publish releases to the tekton-releases GCP project.
+Use tekton standards for code, conduct, and contributions. Choose and document a Tekton stability level for the existing API.
+1. Update the project to make more extensive use of remote resolution. For example, we may want to keep the ability to refer to Tekton CRDs in the same
+repo using local paths, but encourage the use of the built-in hub resolvers rather than using the Pipelines as Code
+[hub resolution](https://pipelinesascode.com/docs/guide/resolver/#tekton-hubhttpshubtektondev) and the built-in git resolver rather than using
+the Pipelines as Code [remote URL syntax](https://pipelinesascode.com/docs/guide/resolver/#remote-http-url).
+(Note: this work is already planned for Pipelines as Code.)
+1. Create a Workflow CRD that allows users to configure much of the behavior that's currently configured through annotations, such as which events
+trigger a PipelineRun and concurrency controls to apply. We may choose to rework the existing implementation so that configuring a PipelineRun
+with annotations would be implemented by creating a Workflow under the hood. We'd support the existing API (PipelineRun with annotations) for at
+least 6 months, and then we can reevaluate how to reduce duplication and create a more consistent API while still maintaining an easy setup experience
+for users who want to configure workflows only via their Git repo rather than on the Kubernetes cluster.
+1. Set a long term goal of updating the project to use Triggers. The project used to be based on Triggers (for more info, see the 
+[latest branch](https://github.com/openshift-pipelines/pipelines-as-code/tree/release-0.5.0/config) using a Triggers backend),
+but migrated to a custom implementation due to several of the [pain points](#use-of-triggers) identified above.
+We'll have to fix these pain points before reworking the implementation to use Triggers, and the refactoring may be challenging to do incrementally
+or not worth the time required to do so. We'll have to weigh the difficulty of refactoring against the difficulty of maintaining
```

Comment:
is weighing the difficulty a requirement to choose this solution ?

Reply:
Maybe. If we think moving PAC to using a Triggers based implementation is so difficult that we will never accomplish it, we should consider if PAC and Triggers functionality/goals are overlapping enough to be confusing to users or difficult to maintain. As you've pointed out, Triggers is much more flexible than PAC, so it's possible the answer is "no".